ha/core/controllers/pcs/pcs_controller.py

```diff
@@ -24,4 +32,76 @@ def __init__(self):
         """
         Initalize pcs controller
         """
-        super(PcsController, self).__init__()
\ No newline at end of file
+        super(PcsController, self).__init__()
+        self._execute = SimpleCommand()
+
+    def heal_resource(self, node_id):
+        """
+        Heal the resources if there are any fail count exists
+        """
+        count = 0
+        while True:
+            time.sleep(10)
+            fail_count_exists = self.check_resource_failcount(node_id)
+            if count >= const.RETRY_COUNT:
+                break
+            if fail_count_exists:
+                self.clean_failure_count(node_id)
+            count += 1
+        return fail_count_exists
+
+    def check_resource_failcount(self, node_id) -> bool:
+        """
+        Resource fail count check
+        """
+        _output, _err, _rc = self._execute.run_cmd(const.PCS_NODE_FAILCOUNT_STATUS.replace("<node>", node_id),
+                                                   check_error=False)
+        if const.NO_FAILCOUNT in _output:
+            return False
+        else:
+            return True
+
+    def clean_failure_count(self, node_id):
+        """
+        Cleanup resources fail count
+        """
+        _output, _err, _rc = self._execute.run_cmd(const.PCS_NODE_CLEANUP.replace("<node>", node_id),
+                                                   check_error=False)
+
+    def nodes_status(self, nodeids: list) -> dict:
+        """
+        Get pcs status of nodes.
+        Args:
+            nodeids (list): List of Node IDs from cluster nodes.
+                Default provide list of all node status.
+                if 'local' then provide local node status.
+
+        Returns:
```

Comment:
Looks like there is a mismatch between returned dict format and mentioned in the docstrings

Reply:
Done changed doc string

Reply:
Accepted